x-pack/plugins/observability_solution/apm/public/utils/http_status_code_to_color.ts

```diff
@@ -5,35 +5,17 @@
  * 2.0.
  */
 
-import { euiLightVars as theme } from '@kbn/ui-theme';
-const { euiColorDarkShade, euiColorWarning } = theme;
+import { useEuiTheme } from '@elastic/eui';
 
-export const errorColor = '#c23c2b';
-export const neutralColor = euiColorDarkShade;
-export const successColor = '#327a42';
-export const warningColor = euiColorWarning;
-
-const httpStatusCodeColors: Record<string, string> = {
-  1: neutralColor,
-  2: successColor,
-  3: neutralColor,
-  4: warningColor,
-  5: errorColor,
+export const useGetStatusColor = (status: string | number) => {
+  const { euiTheme } = useEuiTheme();
+  const httpStatusCodeColors: Record<string, string> = {
+    1: euiTheme.colors.textSubdued,
+    2: euiTheme.colors.textSuccess,
+    3: euiTheme.colors.textSubdued,
+    4: euiTheme.colors.textWarning,
+    5: euiTheme.colors.textDanger,
```

Comment:
I think these are the colors for the badge background not the text, can you try using the following instead:

```suggestion
    1: euiTheme.colors.backgroundBaseSubdued,
    2: euiTheme.colors.backgroundBaseSuccess,
    3: euiTheme.colors.backgroundBaseSubdued,
    4: euiTheme.colors.backgroundBaseWarning,
    5: euiTheme.colors.backgroundBaseDanger,
```

Reply:
<img width="704" alt="Screenshot 2024-12-10 at 10 59 44" src="https://github.com/user-attachments/assets/35a62b3c-c728-4e3e-bf49-d0a7277da71a">
<img width="625" alt="Screenshot 2024-12-10 at 10 59 38" src="https://github.com/user-attachments/assets/5cd87407-e104-4e97-bc4e-2bc991d86cef">
<img width="724" alt="Screenshot 2024-12-10 at 10 59 26" src="https://github.com/user-attachments/assets/e69a32aa-6a32-4e99-8181-cf4eecde872f">
<img width="806" alt="Screenshot 2024-12-10 at 10 59 19" src="https://github.com/user-attachments/assets/d42cb540-b64f-4aba-ab2a-ca7a6343be9e">
This looks much better! thank you.